x-pack/plugins/maps/public/elasticsearch_geo_utils.test.js

```diff
@@ -594,3 +597,125 @@ describe('extractFeaturesFromFilters', () => {
     expect(extractFeaturesFromFilters([spatialFilter])).toEqual([]);
   });
 });
+
+describe('make[Sized]GeotileGridDsl', () => {
+  const field = 'foobar';
+  const bounds = {
+    minLon: 10,
+    maxLon: 20,
+    minLat: 0,
+    maxLat: 1,
+  };
+  const precision = 8;
+
+  it('Should create a dsl with converted bbox', () => {
+    const dsl = makeGeotileGridDsl(field, bounds, precision);
+    expect(dsl).toEqual({
+      bounds: { bottom_right: [20, 0], top_left: [10, 1] },
+      field: 'foobar',
+      precision: 8,
+    });
+  });
+
+  it('Should create a dsl with converted bbox and size', () => {
+    const dsl = makeSizedGeotileGridDsl(field, bounds, precision, 800);
+    expect(dsl).toEqual({
+      bounds: { bottom_right: [20, 0], top_left: [10, 1] },
+      field: 'foobar',
+      precision: 8,
+      size: 800,
+    });
+  });
+});
+
+describe('makeESBbox', () => {
+  it('Should invert Y-axis', () => {
+    const bbox = makeESBbox({
+      minLon: 10,
+      maxLon: 20,
+      minLat: 0,
+      maxLat: 1,
+    });
+    expect(bbox).toEqual({ bottom_right: [20, 0], top_left: [10, 1] });
```

Comment:
Is this correct? Should the top be smaller then bottom?

Reply:
It's weird ES-specific naming convention that mix-and-matches conventions from 2D-graphics APIS (where Y-axis orientation is top to bottom) with conventions from geo-graphic projections (where most projections are North-top/South-bottom). So top should be bigger than bottom, because north-latitude is larger. That's the inverse of things like SVG/HTML Canvas 2D/...